pkg/cmd/cli/cmd/expose.go

```diff
@@ -56,7 +56,13 @@ func NewCmdExpose(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.C
 		cmdutil.CheckErr(err)
 		defRun(cmd, args)
 	}
-	cmd.Flags().String("hostname", "", "Set a hostname for the new route")
+	cmd.Flags().String("hostname", "", "Set a hostname for the new route. Optional.")
+	cmd.Flags().String("path", "", "Path that the router watches to route traffic to the service. Optional.")
+	cmd.Flags().String("tls-termination", "", "Indicate a termination type for the generated route. If not set, any termination config will be ignored. Supports 'edge', 'passthrough', or 'reencrypt'.")
+	cmd.Flags().String("cert-file", "", "Provide certificate contents for a secure route. Used by edge and re-encryption terminated routes.")
```

Comment:
> I don't think this problem starts from the generators.

I'm pretty sure this a problem with generators.  You're going to want this to match ingress capabilities at some point, but now you've got generically named flags that don't actually work in all cases. Services won't do anything with these.